api/v1alpha1/tinkerbell/hardware.go

```diff
@@ -193,6 +196,17 @@ type IP struct {
 	Family  int64  `json:"family,omitempty"`
 }
 
+// ClasslessStaticRoute represents a classless static route for DHCP option 121 (RFC 3442).
+type ClasslessStaticRoute struct {
+	// DestinationDescriptor is the network address and prefix length.
+	// The format is "network/prefixlength", e.g., "192.168.1.0/24" or "10.0.0.0/8".
+	// +kubebuilder:validation:Pattern=`^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)/(3[0-2]|[12]?[0-9])$`
+	DestinationDescriptor string `json:"destination_descriptor,omitempty"`
+	// Router is the IP address of the router for this route.
+	// +kubebuilder:validation:Pattern=`^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$`
+	Router string `json:"router,omitempty"`
```

Comment:
Do these kubebuilder validation patterns prohibit an empty string from being provided? Is there a way to require both fields to be provided if any of them are a non-empty string?

Reply:
i have removed `omitempty`, so now in the CRD you can see they are both required. Also, with the validation of the regex we have, if you provide an empty string, then it will fail due to not matching the regex.